File: openmp/testsuite/fortran/omp_task_private.f
<ompts:test>
<ompts:testdescription>Test which checks the private clause of the task directive. We create a set of tasks in a single region. We defines a variable named sum which gets declared private for each task. Now each task calcualtes a sum using this private variable. Before each calcualation step we introduce a flush command so that maybe the private variabel gets bad. At the end we check if the calculated sum was right.</ompts:testdescription>
<ompts:ompversion>3.0</ompts:ompversion>
<ompts:directive>omp task private</ompts:directive>
<ompts:dependences>omp single,omp critical</ompts:dependences>
<ompts:testcode>
      INCLUDE "omp_my_sleep.f"

      INTEGER FUNCTION <ompts:testcode:functionname>omp_task_firstprivate</ompts:testcode:functionname>()
        IMPLICIT NONE
        INCLUDE "omp_testsuite.f"
        INTEGER j,i
        <ompts:orphan:vars>
        external my_sleep
        INTEGER my_sum
        INTEGER known_sum
        INTEGER rslt
        COMMON /orphvars/ my_sum, known_sum, rslt
        </ompts:orphan:vars>

        my_sum = 0
        rslt = 0
        known_sum = (LOOPCOUNT * (LOOPCOUNT + 1)) / 2

!$omp parallel private(j)
!$omp single
        do i=1, NUM_TASKS
        <ompts:orphan>
!$omp task <ompts:check>private(my_sum)</ompts:check> shared(rslt, known_sum)
          <ompts:check>my_sum = 0</ompts:check>
          do j = 0, LOOPCOUNT
            my_sum = my_sum + j
          end do

          ! check if calculated my_sum was right
          if (my_sum .ne. known_sum) then
!$omp critical
            rslt = rslt + 1
!$omp end critical
          end if
!$omp end task
        </ompts:orphan>
        end do
!$omp end single
!$omp end parallel

        if (rslt .eq. 0) then
            <testfunctionname></testfunctionname> = 1
        else
            <testfunctionname></testfunctionname> = 0
        end if

      END FUNCTION
</ompts:testcode>
</ompts:test>

File: openmp/testsuite/c/omp_task_private.c
<ompts:test>
<ompts:testdescription>Test which checks the private clause of the task directive. We create a set of tasks in a single region. We defines a variable named sum which gets declared private for each task. Now each task calcualtes a sum using this private variable. Before each calcualation step we introduce a flush command so that maybe the private variabel gets bad. At the end we check if the calculated sum was right.</ompts:testdescription>
<ompts:ompversion>3.0</ompts:ompversion>
<ompts:directive>omp task private</ompts:directive>
<ompts:dependences>omp single,omp flush,omp critical</ompts:dependences>
<ompts:testcode>
#include <stdio.h>
#include <math.h>
#include "omp_testsuite.h"

/* Utility function do spend some time in a loop */
int <ompts:testcode:functionname>omp_task_private</ompts:testcode:functionname> (FILE * logFile)
{
    int i;
    <ompts:orphan:vars>
    int known_sum;
    int sum = 0;
    int result = 0; /* counts the wrong sums from tasks */
    </ompts:orphan:vars>

    known_sum = (LOOPCOUNT * (LOOPCOUNT + 1)) / 2;

#pragma omp parallel
    {
#pragma omp single
        {
            for (i = 0; i < NUM_TASKS; i++)
            {
                <ompts:orphan>
#pragma omp task <ompts:check>private(sum)</ompts:check> shared(result, known_sum)
                {
                    int j;
		    //if sum is private, initialize to 0
		    <ompts:check>sum = 0;</ompts:check>
                    for (j = 0; j <= LOOPCOUNT; j++) {
#pragma omp flush
                        sum += j;
                    }
                    /* check if calculated sum was right */
                    if (sum != known_sum) {
#pragma omp critical 
                        result++;
                    }
                } /* end of omp task */
                </ompts:orphan>
            }	/* end of for */
        } /* end of single */
    }	/* end of parallel*/

    return (result == 0);
}
</ompts:testcode>
</ompts:test>
